src/_utils/_git_secret_tools.sh

```diff
@@ -628,8 +628,8 @@ function _get_users_in_gpg_keyring {
 
   # pluck out 'uid' lines, fetch 10th field, extract part in <> if it exists (else leave alone).
   # we use --fixed-list-mode so older versions of gpg emit 'uid:' lines.
-  # sed at the end is to extract email from <>. (If there's no <>, then the line is just an email address anyway.)
-  result=$($SECRETS_GPG_COMMAND "${args[@]}" --no-permission-warning --list-public-keys --with-colon --fixed-list-mode | grep ^uid: | cut -d: -f10 | sed 's/.*<\(.*\)>.*/\1/')
+  # gensub at the end is to extract email from <>. (If there's no <>, then the line is just an email address anyway.)
+  result=$($SECRETS_GPG_COMMAND "${args[@]}" --no-permission-warning --list-public-keys --with-colon --fixed-list-mode | gawk -F: '$1~/uid/{print gensub(/.*<(.*)>.*/, "\\1", "g", $10); }')
```

Comment:
@simbo1905 can you please add a comment explaining your use of gawk here a little more? 

As a reviewer I'm not familiar enough with gawk to say much more than that I can follow the code and it appears to work.

Reply:
I have made a change to the comment. 

A [wise person once told me](https://codereview.stackexchange.com/a/212095/75693) that "In general, any combination of sed, awk, and grep would be better expressed using just an AWK script." So I have been trying to become more familiar with it. Across the git-secret codebase I see examples of using awk and gawk. One is usually symlinked to another. When it comes to doing specific computations using the more specific version seems to make sense so I went with gawk here.

Reply:
Not sure I agree about always favoring `(g)awk` over `grep/cut/sed`, sure we gain some speed and save some keystrokes, but at what cost to readability? How many languages do developers need to know just to make small changes to git-secret?

That having all been said, I'm good with this change if we think it's logically correct.

Reply:
This PR does use some gawk constucts I've never seen before too, though. Do we know if we need a particular minimum version of gawk to support this use?

Reply:
No I am not aware of a minimum version.

Reply:
I was really implying we should make sure we didn't need _too_ recent a version of `gawk` to support `git-secret`. Also it would be good if we could tell users a minimum required version.

Reply:
It's not a big enough deal to worry about if you're not inclined though. LGTM.